### PR TITLE
Math & Client side stuff

### PR DESCRIPTION
**Pull request by BoogieMonster1O1**
stolen from yarn ™ and legacy rewoven yarn ™

### CODE REVIEW

**Comment by BoogieMonster1O1** [comment]
image
slightly brings up the stats

File: mappings/net/minecraft/client/render/BaseFrustum.mapping [hunk 1]
```diff
@@ -0,0 +1,12 @@
+CLASS net/minecraft/class_84 net/minecraft/client/render/BaseFrustum
+	FIELD field_285 frustum [[F
+	FIELD field_286 projectionMatrix [F
+	FIELD field_287 modelViewMatrix [F
```
Comment (valoeghese) [inline mappings/net/minecraft/client/render/BaseFrustum.mapping:4]: couldn’t this just be viewMatrix? Or is this the model transform? In which case something else

File: mappings/net/minecraft/client/render/Frustum.mapping [hunk 2]
```diff
@@ -0,0 +1,9 @@
+CLASS net/minecraft/class_598 net/minecraft/client/render/Frustum
+	FIELD field_2594 instance Lnet/minecraft/class_598;
+	FIELD field_2595 projectionBuffer Ljava/nio/FloatBuffer;
+	FIELD field_2596 modelViewBuffer Ljava/nio/FloatBuffer;
+	METHOD method_1973 getInstance ()Lnet/minecraft/class_84;
+	METHOD method_1974 normalize ([[FI)V
```
Comment (valoeghese) [inline mappings/net/minecraft/client/render/Frustum.mapping:6]: normalise 🇬🇧

File: mappings/net/minecraft/entity/LivingEntity.mapping [hunk 3]
```diff
@@ -22,6 +22,12 @@ CLASS net/minecraft/class_127 net/minecraft/entity/LivingEntity
 	METHOD method_917 getItemTexturePosition (Lnet/minecraft/class_31;)I
 	METHOD method_918 playAmbientSound ()V
 	METHOD method_919 onSpawnedFromSpawner ()V
+	METHOD method_926 getRotationVector (F)Lnet/minecraft/class_26;
+		ARG 1 delta
+	METHOD method_929 raycast (DF)Lnet/minecraft/class_27;
+		ARG 1 distance
+	METHOD method_931 getPositionVector (F)Lnet/minecraft/class_26;
```
Comment (valoeghese) [inline mappings/net/minecraft/entity/LivingEntity.mapping:29]: should these be getRotation and getPos or does that conflict/confuse it with other methods

File: mappings/net/minecraft/entity/Entity.mapping [hunk 4]
```diff
@@ -49,6 +49,7 @@ CLASS net/minecraft/class_57 net/minecraft/entity/Entity
 	METHOD method_1317 baseTick ()V
 	METHOD method_1318 isAlive ()Z
 	METHOD method_1319 destroy ()V
+	METHOD method_1320 getRotationVector ()Lnet/minecraft/class_26;
```
Comment (valoeghese) [inline mappings/net/minecraft/entity/Entity.mapping:52]: Same comment as on living entity

File: mappings/net/minecraft/util/maths/Vec3d.mapping [hunk 5]
```diff
@@ -0,0 +1,58 @@
...
+		ARG 3 y
+		ARG 5 z
+	METHOD method_1292 resetCount ()V
+	METHOD method_1293 create (DDD)Lnet/minecraft/class_26;
+		ARG 0 x
+		ARG 2 y
+		ARG 4 z
+	METHOD method_1294 getDistance (Lnet/minecraft/class_26;)D
+		ARG 1 to
+	METHOD method_1295 interpolateWrtX (Lnet/minecraft/class_26;D)Lnet/minecraft/class_26;
```
Comment (valoeghese) [inline mappings/net/minecraft/util/maths/Vec3d.mapping:18]: What does wrt mean

File: mappings/net/minecraft/util/maths/Vec3d.mapping [hunk 6]
```diff
@@ -0,0 +1,58 @@
...
+		ARG 1 vector
+		ARG 2 x
+	METHOD method_1296 normalize ()Lnet/minecraft/class_26;
```
Comment (valoeghese) [inline mappings/net/minecraft/util/maths/Vec3d.mapping:21]: normalise

File: mappings/net/minecraft/util/maths/Vec3d.mapping [hunk 7]
```diff
@@ -0,0 +1,58 @@
...
+	METHOD method_1297 getOrCreate (DDD)Lnet/minecraft/class_26;
+		COMMENT Returns a cached Vec3f
+		ARG 0 x
+		ARG 2 y
+		ARG 4 z
+	METHOD method_1298 getDistanceSquared (Lnet/minecraft/class_26;)D
```
Comment (valoeghese) [inline mappings/net/minecraft/util/maths/Vec3d.mapping:27]: Name debate time: getDistanceSquared vs sqrDist 😎

Comment (valoeghese) [reply mappings/net/minecraft/util/maths/Vec3d.mapping:27]: One is concise but the other is more readable

**Review by valoeghese — CHANGES_REQUESTED** [review]
Mostly nitpicks and questions

File: mappings/net/minecraft/util/maths/Vec3d.mapping (hunk 5 above)
Comment (BoogieMonster1O1) [reply mappings/net/minecraft/util/maths/Vec3d.mapping:18]: with respect to

Comment (BoogieMonster1O1) [reply mappings/net/minecraft/util/maths/Vec3d.mapping:18]: here you give it a vec3d and an x coordinate and it generates a new vec3d with said x coordinate

Comment (valoeghese) [reply mappings/net/minecraft/util/maths/Vec3d.mapping:18]: Weird code but ok

**Review by valoeghese — APPROVED** [review]
image
How did we let this in lmao

Ok epic approv time to last call this

**Comment by BoogieMonster1O1** [comment]
> image How did we let this in lmao
> 
> Ok epic approv time to last call this

blurted

**Comment by BoogieMonster1O1** [comment]
my pull request titles shall not be anglicized

**Review by thecatcore — APPROVED** [review]

**Pull request merged**